packages/ast-spec/src/type/TSQualifiedName/spec.ts

```diff
@@ -1,10 +1,11 @@
 import type { AST_NODE_TYPES } from '../../ast-node-types';
 import type { BaseNode } from '../../base/BaseNode';
 import type { Identifier } from '../../expression/Identifier/spec';
+import type { PrivateIdentifier } from '../../special/PrivateIdentifier/spec';
 import type { EntityName } from '../../unions/EntityName';
 
 export interface TSQualifiedName extends BaseNode {
   type: AST_NODE_TYPES.TSQualifiedName;
   left: EntityName;
-  right: Identifier;
+  right: Identifier | PrivateIdentifier;
```

Comment:
NGL - I do hate this a little bit because the only place that `this.#foo` is valid is in a `typeof`. It just means that consumers will have to check the type of this even though they'll know it's not possible.

It's probably not worth going to the effort of creating a separate AST node just for private member expressions within a type query.